src/main/java/org/tugraz/sysds/runtime/data/HeterogTensor.java

```diff
@@ -0,0 +1,383 @@
+/*
+ * Copyright 2019 Graz University of Technology
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.tugraz.sysds.runtime.data;
+
+import org.apache.commons.lang.ArrayUtils;
+import org.tugraz.sysds.common.Types.ValueType;
+import org.tugraz.sysds.runtime.DMLRuntimeException;
+import org.tugraz.sysds.runtime.controlprogram.caching.CacheBlock;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.Arrays;
+
+public class HeterogTensor extends TensorBlock {
+	private HomogTensor[] _colsdata = null;
+	private ValueType[] _schema = null;
```

Comment:
So you're using a single tensor per column - please add a TODO comment to reconsider using a single tensor for each value type.

Reply:
Sorry, I completely forgot that we talked about the implementation with only one `BasicTensor` for each datatype.